Feature/task1/impl

Pavel-Laukhin opened the pull request:
Все сделал, и даже больше: при попытке добавить элемент с дублированным именем, он получит имя с числовым суффиксом в круглых скобках. 

Например, была папка "Folder". При попытке добавить такую же, появится папка "Folder(1)", затем "Folder(2)" и так далее.

Единственное, я не делал специальной обработки имён файлов, чтобы суффикс появлялся перед расширением. В целом, это не сложно, просто не было такой цели, ведь файлы в этом задании, как я понял, создаются без расширения.

Однако есть проблема, которую я не знаю, как побороть. Не получается добавить элемент с тем же именем, но с другим регистром у первой буквы (другие не проверял). Например, если есть папка "Folder", то не получается добавить ни папку, ни файл с именем "folder". Прям печаль. Весь гугл излазил, но так и не нашел, в чем причина (((

Как это исправить?

Pavel-Laukhin committed:
Template is ready

Pavel-Laukhin committed:
FileManagerService has been added...

Table view now provides deletion and adding folders.
A tap on a folder leads to new view controller of the folder and its content.

Pavel-Laukhin committed:
1. Files now also could be added...

2. ViewController now takes a folder path to be created.
3. Added case-insensitive sorting of the list of folders and files, as well as case-insensitive insertion.
4. Added FileContentViewController for the ability to represent the content of a file.

Pavel-Laukhin committed:
Fixed a bug when files created as folders...

That required creating a new method of getting URL for item in FileManagerService.
Also changed some names to more appropriate.

Pavel-Laukhin committed:
Now you can read and delete items in subfolders.

Pavel-Laukhin committed:
Now new elements with previously registered names...

... get an index at the end of their name.

Pavel-Laukhin committed:
insignificant change

Pavel-Laukhin committed:
Got fid of extra code

Pavel-Laukhin committed:
Got rid of the warnings+edited the files structure

MaksimSoldatov commented on File Manager/FileManagerService/FileManagerService.swift:
```diff
@@ -0,0 +1,105 @@
+        
+    enum AppDirectories: String, CaseIterable {
+        case Documents = "Documents"
+        case Inbox = "Inbox"
```
Неиспользуемые директории

MaksimSoldatov commented on File Manager/FileManagerService/FileManagerService.swift:
```diff
@@ -0,0 +1,105 @@
+    /// Добавляет папку. Возвращает true в случае удачи и false в случае неудачи.
+    func addFolderNamed(as name: String, in directory: String) -> Bool
+    
+    /// Добавляет файл. Возвращает true в случае удачи и false в случае неудачи.
```
Для чего столько комментов?

Pavel-Laukhin replied:
Исправил

MaksimSoldatov commented on File Manager/AppDelegate.swift:
```diff
@@ -14,13 +14,11 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        
-        let mainVC = ViewController()
```
Нэйминг

Pavel-Laukhin replied:
Переименовал на BaseViewController и baseVC соответственно.

MaksimSoldatov commented on File Manager/FileManagerService/FileManagerService.swift:
```diff
@@ -88,7 +87,7 @@ struct FileManagerService: FileManagerServiceProtocol {
                 return nil
             }
         } else {
-            // Переименовываем задублированную папку. Добавляем к имени индекс до тех пор, пока новое имя не станет уникальным. После этого возвращаем новое имя.
```
Лучше вынести в отдельный метод, тогда не придется писать коммент.

Pavel-Laukhin replied:
Избавился от рекурсии и вынес в отдельный метод.

Pavel-Laukhin committed:
Fixed issues

1. Renamed some entities
2. Got rid of recursion method and properties that were used for it. Now there is a separate method instead.
